.github/workflows/build-spack.yml

```diff
@@ -135,15 +135,38 @@ jobs:
             test -n "${G4LEDATA}"
           fi
           cmake --preset=${CMAKE_PRESET} --log-level=VERBOSE
+      - name: Run incremental clang-tidy
+        id: clang-tidy
+        if: ${{matrix.special == 'clang-tidy'}}
+        env:
+          BASE_REF: "${{format('{0}', github.base_ref || 'develop')}}"
+        run: |
+          if [ "${{github.event_name}}" == "schedule" ]; then
+            echo "Full clang-tidy check on scheduled run."
+            ninja -k0
+            exit $?
+          fi
+          set -x
+          FIRST_PR_HASH=$(git log --reverse --oneline --format="%H" origin/${BASE_REF}..HEAD | head -1)
```

Comment:
Does this work if it has a fetch depth of zero?

Reply:
And related question, isn't there a github action variable that already has this information?

Reply:
it should, fetch-depth of 0 means fetch all the history. I reproduced the commands that the checkout action uses to clone the repo locally and it works, but not on the runner...

Reply:
Oh, it's not the "shallowest". Actually I think you may want to use the "destination branch" sha; the GHA always runs on the result of merging the PR into the target. So usually `develop` *would* be correct—sorry I forgot that earlier.

Reply:
> And related question, isn't there a github action variable that already has this information?

There are way to get the source/destination branch or commit that triggered the workflow but I haven't seen a variable with the base commit of a PR.




> Oh, it's not the "shallowest". Actually I think you may want to use the "destination branch" sha; the GHA always runs on the result of merging the PR into the target. So usually `develop` _would_ be correct—sorry I forgot that earlier.

But that would include any change now in develop that not present in that PR, from the [action logs](https://github.com/celeritas-project/celeritas/actions/runs/12169546804/job/33942867739?pr=1531), I get the correct list of modified files, it seems that grep is failing. 🤔